File: src/Modal/Prog7/Eigen2DOF/CFrame.java
package Modal.Prog7.Eigen2DOF;

import java.awt.Graphics;
import java.awt.Point;
import java.awt.Rectangle;

@SuppressWarnings("serial")
class CFrame extends Rectangle {
   public static final int kMouseDown = 0;
   public static final int kMouseDrag = 1;
   public static final int kMouseUp = 2;
   public static final int kMouseMove = 3;
   public boolean mWasHit;
   CFramePanel mFramePanel;

   public CFrame(CFramePanel var1) {
      super(0, 0, 0, 0);
      this.mFramePanel = var1;
      this.mFramePanel.AddFrame(this);
   }

   public CFrame(CFramePanel var1, int var2, int var3, int var4, int var5) {
      super(var2, var3, var4, var5);
      this.mFramePanel = var1;
      this.mFramePanel.AddFrame(this);
   }

   public void Frame(Graphics var1) {
   }

   public boolean MouseEvent(int var1, boolean var2) {
      this.mWasHit = this.contains(this.mFramePanel.mThisPt.x, this.mFramePanel.mThisPt.y);
      return false;
   }

   public Point GlobalToLocal(Point var1) {
      return var1 == null ? null : new Point(var1.x - super.x, var1.y - super.y);
   }

   public Rectangle GlobalToLocalR(Rectangle var1) {
      return var1 == null ? null : new Rectangle(var1.x - super.x, var1.y - super.y, var1.width, var1.height);
   }

   public Point LocalToGlobal(Point var1) {
      return var1 == null ? null : new Point(var1.x + super.x, var1.y + super.y);
   }

   public Rectangle LocalToGlobalR(Rectangle var1) {
      return var1 == null ? null : new Rectangle(var1.x + super.x, var1.y + super.y, var1.width, var1.height);
   }

   public void repaint() {
      this.mFramePanel.repaint();
   }

   public void repaint(long var1) {
      this.mFramePanel.repaint(var1);
   }

   public void ControlMessage(int var1, double var2) {
      this.ControlMessage(null, var1, var2);
   }

   public void ControlMessage(CFrame var1, int var2, double var3) {
   }
}

File: site/swingjs/j2s/TwoDOF/Abutmentsin_2DOF/Forced/CFrame.js
(function(){var P$=Clazz.newPackage("TwoDOF.Abutmentsin_2DOF.Forced"),I$=[[0,'java.awt.Point','java.awt.Rectangle']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "CFrame", null, 'java.awt.Rectangle');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['Z',['mWasHit'],'O',['mFramePanel','TwoDOF.Abutmentsin_2DOF.Forced.CFramePanel']]]

Clazz.newMeth(C$, 'c$$TwoDOF_Abutmentsin_2DOF_Forced_CFramePanel',  function (var1) {
;C$.superclazz.c$$I$I$I$I.apply(this,[0, 0, 0, 0]);C$.$init$.apply(this);
this.mFramePanel=var1;
this.mFramePanel.AddFrame$TwoDOF_Abutmentsin_2DOF_Forced_CFrame(this);
}, 1);

Clazz.newMeth(C$, 'c$$TwoDOF_Abutmentsin_2DOF_Forced_CFramePanel$I$I$I$I',  function (var1, var2, var3, var4, var5) {
;C$.superclazz.c$$I$I$I$I.apply(this,[var2, var3, var4, var5]);C$.$init$.apply(this);
this.mFramePanel=var1;
this.mFramePanel.AddFrame$TwoDOF_Abutmentsin_2DOF_Forced_CFrame(this);
}, 1);

Clazz.newMeth(C$, 'Frame$java_awt_Graphics',  function (var1) {
});

Clazz.newMeth(C$, 'MouseEvent$I$Z',  function (var1, var2) {
this.mWasHit=this.contains$I$I(this.mFramePanel.mThisPt.x, this.mFramePanel.mThisPt.y);
return false;
});

Clazz.newMeth(C$, 'GlobalToLocal$java_awt_Point',  function (var1) {
return var1 == null  ? null : Clazz.new_($I$(1,1).c$$I$I,[var1.x - this.x, var1.y - this.y]);
});

Clazz.newMeth(C$, 'GlobalToLocalR$java_awt_Rectangle',  function (var1) {
return var1 == null  ? null : Clazz.new_($I$(2,1).c$$I$I$I$I,[var1.x - this.x, var1.y - this.y, var1.width, var1.height]);
});

Clazz.newMeth(C$, 'LocalToGlobal$java_awt_Point',  function (var1) {
return var1 == null  ? null : Clazz.new_($I$(1,1).c$$I$I,[var1.x + this.x, var1.y + this.y]);
});

Clazz.newMeth(C$, 'LocalToGlobalR$java_awt_Rectangle',  function (var1) {
return var1 == null  ? null : Clazz.new_($I$(2,1).c$$I$I$I$I,[var1.x + this.x, var1.y + this.y, var1.width, var1.height]);
});

Clazz.newMeth(C$, 'repaint$',  function () {
this.mFramePanel.repaint$();
});

Clazz.newMeth(C$, 'repaint$J',  function (var1) {
this.mFramePanel.repaint$J(var1);
});

Clazz.newMeth(C$, 'ControlMessage$I$D',  function (var1, var2) {
this.ControlMessage$TwoDOF_Abutmentsin_2DOF_Forced_CFrame$I$D(null, var1, var2);
});

Clazz.newMeth(C$, 'ControlMessage$TwoDOF_Abutmentsin_2DOF_Forced_CFrame$I$D',  function (var1, var2, var3) {
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:51 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
